GitHub actions

rportilla-databricks opened the pull request:
1. updating project to use github action for running tests 
2. updated to publish to codecov

tnixon reviewed (changes requested):
There seems to be a number of files generated by the code coverage tool. We should not save generated files in git repo - let's add these files & folders to `.gitignore`

rportilla-databricks commented:
> There seems to be a number of files generated by the code coverage tool. We should not save generated files in git repo - let's add these files & folders to `.gitignore`

Good call, I've removed and added to .gitignore.
